hstrat/genome_instrumentation/_HereditaryStratigraphicColumn.py

```diff
@@ -96,12 +103,24 @@ def __init__(
         """
         self._always_store_rank_in_stratum = always_store_rank_in_stratum
         self._stratum_differentia_bit_width = stratum_differentia_bit_width
-        self._num_strata_deposited = 0
-        self._stratum_ordered_store = stratum_ordered_store_factory()
-
         self._stratum_retention_policy = stratum_retention_policy
 
-        self.DepositStratum(annotation=initial_stratum_annotation)
+        if stratum_ordered_store_factory is None:
+            # if no hstrat ordered store is specified, we use a list
+            stratum_ordered_store_factory = HereditaryStratumOrderedStoreList
+        if callable(stratum_ordered_store_factory):
+            # ordered store is actually an ordered store factory
+            self._stratum_ordered_store = stratum_ordered_store_factory()
+            self._num_strata_deposited = 0
+            self.DepositStratum(annotation=initial_stratum_annotation)
+        elif isinstance(
+            stratum_ordered_store_factory[0], HereditaryStratumOrderedStoreBase
+        ):
+            # ordered store is already an instance of an ordered store
+            (
+                self._stratum_ordered_store,
+                self._num_strata_deposited,
+            ) = stratum_ordered_store_factory
```

Comment:
Add and else to throw a `ValueError` for bad param or something

Reply:
Sure! 1d299fd